src/hotspot/share/oops/instanceKlass.hpp

```diff
@@ -997,7 +993,7 @@ class InstanceKlass: public Klass {
 
   // The constant pool is on stack if any of the methods are executing or
   // referenced by handles.
-  bool on_stack() const { return _constants->on_stack(); }
+  virtual bool on_stack() const;
```

Comment:
The comment looks weird now that the implementation, which delegates to the constant pool, has been moved from the header. I think the comment needs to be updated.

Reply:
Suggest move the comment to the implementation.